Install script fails on macOS: darwin binaries not built in release pipeline

## Problem Description

The installation script fails on macOS with a tar archive format error because the GitHub Actions release pipeline only builds Linux binaries but the install script expects macOS binaries to be available.

## Error Message

```bash
[localhost:mcp-mesh] $ curl -sSL https://raw.githubusercontent.com/dhyansraj/mcp-mesh/main/install.sh | bash
[INFO] MCP Mesh Binary Installer
[INFO] =========================
[INFO] Detected platform: darwin-arm64
[INFO] Fetching latest version from GitHub...
[INFO] Latest version: v0.5.1
[INFO] Downloading meshctl v0.5.1 for darwin-arm64...
[INFO] URL: https://github.com/dhyansraj/mcp-mesh/releases/download/v0.5.1/mcp-mesh_v0.5.1_darwin_arm64.tar.gz
[INFO] Extracting meshctl...
tar: Error opening archive: Unrecognized archive format
[ERROR] Failed to extract archive
```

## Root Cause Analysis

1. **Install Script Supports macOS**: The `install.sh` script correctly detects `darwin` (macOS) platforms and expects `darwin-arm64.tar.gz` and `darwin-amd64.tar.gz` files
2. **Build Pipeline Missing macOS**: The GitHub Actions release workflow (`.github/workflows/release.yml` line 63) only builds for Linux platforms:
   ```yaml
   PLATFORMS: "linux/amd64,linux/arm64"
   ```
3. **Missing Artifacts**: The expected `mcp-mesh_v0.5.1_darwin_arm64.tar.gz` file doesn't exist in releases because it's never built

## Proposed Solutions

### Option 1: Add macOS Cross-Compilation (Recommended)
Update `.github/workflows/release.yml` line 63:
```yaml
# Current (broken):
PLATFORMS: "linux/amd64,linux/arm64"

# Should be:
PLATFORMS: "linux/amd64,linux/arm64,darwin/amd64,darwin/arm64"
```

**Considerations:**
- The registry binary uses SQLite with CGO, which may require additional cross-compilation setup
- May need to disable CGO for macOS builds or add macOS-specific toolchain

### Option 2: Native macOS Builds
Add a separate GitHub Actions job using `macos-latest` runners to build native macOS binaries:
```yaml
build-macos-binaries:
  runs-on: macos-latest
  steps:
    # Build darwin/amd64 and darwin/arm64 natively
```

### Option 3: Install Script Fallback
Update `install.sh` to detect unsupported platforms and provide alternative installation methods:
```bash
# If binary not available, suggest pip install
pip install mcp-mesh
```

## Impact
- **Severity**: High - Completely blocks macOS users from using the install script
- **Workaround**: Users can install via `pip install mcp-mesh` instead
- **Affected Versions**: All releases that use the current pipeline (likely v0.5.0+)

## Files Involved
- `.github/workflows/release.yml` (line 63) - Build pipeline configuration
- `packaging/scripts/build-binaries.sh` (line 12) - Build script platform defaults
- `install.sh` (lines 120-122) - Platform detection logic

## Testing Steps
1. Fix the pipeline to build macOS binaries
2. Create a test release
3. Verify `mcp-mesh_vX.X.X_darwin_arm64.tar.gz` and `mcp-mesh_vX.X.X_darwin_amd64.tar.gz` are created
4. Test install script on macOS: `curl -sSL https://raw.githubusercontent.com/user/repo/branch/install.sh | bash`

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -15,7 +15,7 @@ on:
         description: "Environment"
         type: choice
         options: [production, test]
-        default: test
+        default: production
 
 env:
   REGISTRY: mcpmesh
@@ -38,6 +38,7 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+          cache: false
 
       - name: Install cross-compilation toolchain
         run: |
@@ -84,10 +85,204 @@ jobs:
             dist/*.zip
             dist/checksums.txt
 
+  # Build macOS binaries natively
+  build-macos-binaries:
+    runs-on: macos-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: false
+
+      - name: Determine version
+        id: version
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            VERSION="${{ github.event.inputs.version }}"
+          elif [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            VERSION="${{ github.ref_name }}"
+          else
+            VERSION="${{ github.sha }}"
+          fi
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+          echo "Building macOS version: ${VERSION}"
+
+      - name: Build macOS binaries
+        env:
+          VERSION: ${{ steps.version.outputs.version }}
+          PLATFORMS: "darwin/amd64,darwin/arm64"
+          OUTPUT_DIR: "dist-macos"
+        run: ./packaging/scripts/build-binaries.sh
+
+      - name: Upload release assets
+        if: github.event_name == 'release'
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            dist-macos/*.tar.gz
+            dist-macos/*.zip
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload artifacts
+        if: github.event_name != 'release'
+        uses: actions/upload-artifact@v4
+        with:
+          name: macos-binaries-${{ steps.version.outputs.version }}
+          path: |
+            dist-macos/*.tar.gz
+            dist-macos/*.zip
+
+  # Combine checksums from all platforms (for real releases)
+  combine-checksums:
+    runs-on: ubuntu-latest
+    needs: [build-binaries, build-macos-binaries]
+    if: github.event_name == 'release'
+    permissions:
+      contents: write
+    steps:
+      - name: Determine version
+        id: version
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            VERSION="${{ github.event.inputs.version }}"
+          elif [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            VERSION="${{ github.ref_name }}"
+          else
+            VERSION="${{ github.sha }}"
+          fi
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+      - name: Download Linux binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: binaries-${{ steps.version.outputs.version }}
+          path: dist-linux
+
+      - name: Download macOS binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: macos-binaries-${{ steps.version.outputs.version }}
+          path: dist-macos
+
+      - name: Generate combined checksums
+        run: |
+          mkdir -p dist-combined
+
+          # Copy all archives to combined directory
+          find dist-linux -name "*.tar.gz" -o -name "*.zip" | xargs -I {} cp {} dist-combined/
+          find dist-macos -name "*.tar.gz" -o -name "*.zip" | xargs -I {} cp {} dist-combined/
+
+          # Generate combined checksums
+          cd dist-combined
+          if command -v sha256sum &> /dev/null; then
+            sha256sum *.tar.gz > checksums.txt
+            if ls *.zip &> /dev/null; then
+              sha256sum *.zip >> checksums.txt
+            fi
+          elif command -v shasum &> /dev/null; then
+            shasum -a 256 *.tar.gz > checksums.txt
+            if ls *.zip &> /dev/null; then
+              shasum -a 256 *.zip >> checksums.txt
+            fi
+          fi
+
+          echo "Combined checksums:"
+          cat checksums.txt
+
+      - name: Upload combined checksums to release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: dist-combined/checksums.txt
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  # Publish Python package to PyPI
+  publish-python:
+    runs-on: ubuntu-latest
+    needs: [build-binaries, build-macos-binaries]
+    if: >
+      github.event_name == 'release' ||
+      (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
+    permissions:
+      contents: read
+      id-token: write # For trusted publishing
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install build dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build twine
+
+      - name: Prepare packaging files
+        working-directory: packaging/pypi
+        run: |
+          cp ../../src/runtime/python/README.md .
+          cp ../../LICENSE .
+          cp -r ../../src/runtime/python/_mcp_mesh .
+          cp -r ../../src/runtime/python/mesh .
+
+      - name: Update version in packaging files
+        working-directory: packaging/pypi
+        run: |
+          # Determine version
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            VERSION="${{ github.event.inputs.version }}"
+          else
+            VERSION="${{ github.ref_name }}"
+          fi
+          VERSION=${VERSION#v}  # Remove 'v' prefix
+
+          # Debug output
+          echo "Event name: ${{ github.event_name }}"
+          echo "Ref name: ${{ github.ref_name }}"
+          echo "Detected version: ${VERSION}"
+          echo "Current version in file: $(grep '__version__' _mcp_mesh/__init__.py)"
+
+          # Update version in both __init__.py and pyproject.toml
+          sed -i "s/__version__ = \".*\"/__version__ = \"${VERSION}\"/" _mcp_mesh/__init__.py
+          sed -i "s/version = \".*\"/version = \"${VERSION}\"/" pyproject.toml
+
+          # Verify the versions were updated
+          echo "Updated __init__.py version to: $(grep '__version__' _mcp_mesh/__init__.py)"
+          echo "Updated pyproject.toml version to: $(grep 'version = ' pyproject.toml)"
+
+      - name: Build package
+        working-directory: packaging/pypi
+        run: python -m build
+
+      - name: Check package
+        working-directory: packaging/pypi
+        run: twine check dist/*
+
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: packaging/pypi/dist/
+          password: ${{ secrets.PYPI_API_TOKEN }}
+
   # Build and push Docker images
   build-docker:
     runs-on: ubuntu-latest
     needs: [publish-python]
+    if: >
+      github.event_name == 'release' ||
+      (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
     permissions:
       contents: read
       packages: write
@@ -99,18 +294,12 @@ jobs:
         uses: docker/setup-buildx-action@v3
 
       - name: Log in to Docker Hub
-        if: >
-          github.event_name == 'release' ||
-          (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKER_HUB_USERNAME }}
           password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
 
       - name: Log in to GitHub Container Registry
-        if: >
-          github.event_name == 'release' ||
-          (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
         uses: docker/login-action@v3
         with:
           registry: ghcr.io
@@ -161,9 +350,7 @@ jobs:
           platforms: linux/amd64,linux/arm64
           build-args: |
             VERSION=${{ steps.meta.outputs.version_with_v }}
-          push: >
-            ${{ github.event_name == 'release' ||
-            (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production') }}
+          push: true
           tags: |
             ${{ env.REGISTRY }}/registry:${{ steps.meta.outputs.patch_version }}
             ${{ env.REGISTRY }}/registry:${{ steps.meta.outputs.minor_version }}
@@ -222,9 +409,7 @@ jobs:
           platforms: linux/amd64,linux/arm64
           build-args: |
             VERSION=${{ steps.meta.outputs.version }}
-          push: >
-            ${{ github.event_name == 'release' ||
-            (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production') }}
+          push: true
           tags: |
             ${{ env.REGISTRY }}/python-runtime:${{ steps.meta.outputs.patch_version }}
             ${{ env.REGISTRY }}/python-runtime:${{ steps.meta.outputs.minor_version }}
@@ -245,9 +430,7 @@ jobs:
           platforms: linux/amd64,linux/arm64
           build-args: |
             VERSION=${{ steps.meta.outputs.version_with_v }}
-          push: >
-            ${{ github.event_name == 'release' ||
-            (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production') }}
+          push: true
           tags: |
             ${{ env.REGISTRY }}/cli:${{ steps.meta.outputs.patch_version }}
             ${{ env.REGISTRY }}/cli:${{ steps.meta.outputs.minor_version }}
@@ -260,89 +443,13 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
 
-  # Publish Python package to PyPI
-  publish-python:
-    runs-on: ubuntu-latest
-    needs: [build-binaries]
-    if: >
-      github.event_name == 'release' ||
-      (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
-    permissions:
-      contents: read
-      id-token: write # For trusted publishing
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ env.PYTHON_VERSION }}
-
-      - name: Install build dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install build twine
-
-      - name: Prepare packaging files
-        working-directory: packaging/pypi
-        run: |
-          cp ../../src/runtime/python/README.md .
-          cp ../../LICENSE .
-          cp -r ../../src/runtime/python/_mcp_mesh .
-          cp -r ../../src/runtime/python/mesh .
-
-      - name: Update version in packaging files
-        working-directory: packaging/pypi
-        run: |
-          # Determine version
-          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
-            VERSION="${{ github.event.inputs.version }}"
-          else
-            VERSION="${{ github.ref_name }}"
-          fi
-          VERSION=${VERSION#v}  # Remove 'v' prefix
-
-          # Convert test versions to PEP 440 compliant format
-          if [[ "$VERSION" == *"-test"* ]]; then
-            VERSION=$(echo "$VERSION" | sed 's/-test/.dev/')
-          fi
-
-          # Debug output
-          echo "Event name: ${{ github.event_name }}"
-          echo "Ref name: ${{ github.ref_name }}"
-          echo "Detected version: ${VERSION}"
-          echo "Current version in file: $(grep '__version__' _mcp_mesh/__init__.py)"
-
-          # Update version in both __init__.py and pyproject.toml
-          sed -i "s/__version__ = \".*\"/__version__ = \"${VERSION}\"/" _mcp_mesh/__init__.py
-          sed -i "s/version = \".*\"/version = \"${VERSION}\"/" pyproject.toml
-
-          # Verify the versions were updated
-          echo "Updated __init__.py version to: $(grep '__version__' _mcp_mesh/__init__.py)"
-          echo "Updated pyproject.toml version to: $(grep 'version = ' pyproject.toml)"
-
-      - name: Build package
-        working-directory: packaging/pypi
-        run: python -m build
-
-      - name: Check package
-        working-directory: packaging/pypi
-        run: twine check dist/*
-
-      - name: Publish to PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          packages-dir: packaging/pypi/dist/
-          password: ${{ secrets.PYPI_API_TOKEN }}
-
   # Update package manager manifests
   update-packages:
     runs-on: ubuntu-latest
+    needs: [build-binaries, build-macos-binaries, combine-checksums]
     if: >
       github.event_name == 'release' ||
       (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
-    needs: [build-binaries]
     permissions:
       contents: write
       pull-requests: write
@@ -351,32 +458,102 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Update Homebrew formula
+        env:
+          GITHUB_TOKEN: ${{ secrets.HOMEBREW_PAT_TOKEN }}
         run: |
           if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
             VERSION="${{ github.event.inputs.version }}"
           else
             VERSION="${{ github.ref_name }}"
           fi
-          VERSION=${VERSION#v}  # Remove 'v' prefix
+          VERSION_NO_V=${VERSION#v}  # Remove 'v' prefix
+          TAG_NAME="$VERSION"
 
-          # Download checksums
-          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
-            TAG_NAME="${{ github.event.inputs.version }}"
-          else
-            TAG_NAME="${{ github.ref_name }}"
-          fi
+          echo "Updating Homebrew formula for version: $VERSION"
 
+          # Download checksums from release
           curl -L \
             "https://github.com/${{ github.repository }}/releases/download/${TAG_NAME}/checksums.txt" \
             -o checksums.txt
 
-          # Update Homebrew formula with new version and checksums
-          # This would typically involve updating the formula file and creating a PR
-          # For now, we'll just log the information needed
-          echo "Homebrew formula update needed:"
-          echo "Version: ${VERSION}"
-          echo "Checksums:"
-          cat checksums.txt
+          # Extract SHA256 for macOS binaries
+          DARWIN_AMD64_SHA=$(grep "mcp-mesh_${TAG_NAME}_darwin_amd64.tar.gz" checksums.txt | cut -d' ' -f1)
+          DARWIN_ARM64_SHA=$(grep "mcp-mesh_${TAG_NAME}_darwin_arm64.tar.gz" checksums.txt | cut -d' ' -f1)
+
+          if [[ -z "$DARWIN_AMD64_SHA" || -z "$DARWIN_ARM64_SHA" ]]; then
+            echo "Error: Could not find macOS checksums in checksums.txt"
+            cat checksums.txt
+            exit 1
+          fi
+
+          echo "Darwin AMD64 SHA: $DARWIN_AMD64_SHA"
+          echo "Darwin ARM64 SHA: $DARWIN_ARM64_SHA"
+
+          # Create updated formula
+          REPO_URL="https://github.com/dhyansraj/mcp-mesh"
+          ARM64_URL="${REPO_URL}/releases/download/${TAG_NAME}/mcp-mesh_${TAG_NAME}_darwin_arm64.tar.gz"
+          AMD64_URL="${REPO_URL}/releases/download/${TAG_NAME}/mcp-mesh_${TAG_NAME}_darwin_amd64.tar.gz"
+
+          cat > mcp-mesh.rb << EOF
+          class McpMesh < Formula
+            desc "Distributed service orchestration framework built on the Model Context Protocol"
+            homepage "${REPO_URL}"
+            url "${ARM64_URL}"
+            sha256 "${DARWIN_ARM64_SHA}"
+            license "MIT"
+            version "${VERSION_NO_V}"
+
+            # Dependencies
+            depends_on "go" => :build
+
+            on_macos do
+              if Hardware::CPU.intel?
+                url "${AMD64_URL}"
+                sha256 "${DARWIN_AMD64_SHA}"
+              end
+            end
+
+            def install
+              # Install meshctl CLI
+              bin.install "meshctl"
+
+              # Install registry binary
+              bin.install "mcp-mesh-registry"
+            end
+
+            def caveats
+              <<~EOS
+                MCP Mesh has been installed with two binaries:
+                  • meshctl         - CLI tool for managing MCP Mesh
+                  • mcp-mesh-registry - Registry service for service discovery
+
+                To start the registry service:
+                  mcp-mesh-registry
+
+                To use the CLI:
+                  meshctl --help
+
+                For more information:
+                  https://github.com/dhyansraj/mcp-mesh/blob/main/README.md
+              EOS
+            end
+
+            test do
+              # Test that the binaries are installed and can show version/help
+              system "#{bin}/meshctl", "--help"
+              system "#{bin}/mcp-mesh-registry", "--help"
+            end
+          end
+          EOF
+
+          # Update the formula in the tap repository
+          gh api --method PUT \
+            "/repos/dhyansraj/homebrew-mcp-mesh/contents/Formula/mcp-mesh.rb" \
+            -f message="Update to ${VERSION}" \
+            -f content="$(base64 -w 0 mcp-mesh.rb)" \
+            -f sha="$(gh api /repos/dhyansraj/homebrew-mcp-mesh/contents/Formula/mcp-mesh.rb --jq '.sha')"
+
+          echo "✅ Homebrew formula updated successfully to ${VERSION}"
 
       - name: Update Scoop manifest
         run: |
@@ -392,41 +569,10 @@ jobs:
 
           echo "Scoop manifest updated for version: ${VERSION}"
 
-  # Security scan - Disabled due to 0.0.0.0 binding causing false positives
-  # security-scan:
-  #   runs-on: ubuntu-latest
-  #   if: >
-  #     github.event_name == 'release' ||
-  #     (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
-  #   needs: [build-docker]
-  #   permissions:
-  #     contents: read
-  #   steps:
-  #     - name: Run Trivy vulnerability scanner on Registry image
-  #       uses: aquasecurity/trivy-action@master
-  #       with:
-  #         image-ref: "${{ env.REGISTRY }}/registry:latest"
-  #         format: "table"
-  #         exit-code: "0" # Don't fail on vulnerabilities for now
-  #
-  #     - name: Run Trivy vulnerability scanner on Python Runtime image
-  #       uses: aquasecurity/trivy-action@master
-  #       with:
-  #         image-ref: "${{ env.REGISTRY }}/python-runtime:latest"
-  #         format: "table"
-  #         exit-code: "0"
-  #
-  #     - name: Run Trivy vulnerability scanner on CLI image
-  #       uses: aquasecurity/trivy-action@master
-  #       with:
-  #         image-ref: "${{ env.REGISTRY }}/cli:latest"
-  #         format: "table"
-  #         exit-code: "0"
-
   # Discord notification for successful releases
   notify-discord:
     runs-on: ubuntu-latest
-    needs: [build-binaries, publish-python, build-docker, update-packages]
+    needs: [build-binaries, build-macos-binaries, publish-python, build-docker, update-packages]
     if: >
       ${{ success() && (github.event_name == 'release' ||
       (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')) }}
@@ -446,12 +592,23 @@ jobs:
       - name: Discord Webhook Notification
         run: |
           VERSION="${{ steps.version.outputs.version }}"
-          RELEASE_URL="${{ github.event.release.html_url || format('https://github.com/{0}/releases/tag/{1}', github.repository, steps.version.outputs.version) }}"
+          RELEASE_URL="${{ github.event.release.html_url }}"
+          if [[ -z "$RELEASE_URL" ]]; then
+            RELEASE_URL="https://github.com/${{ github.repository }}/releases/tag/${{ steps.version.outputs.version }}"
+          fi
 
           # Create JSON payload for Discord webhook with thread_name for forum channels
+          CONTENT="🚀 **New MCP Mesh Release!**\\n\\n**Version:** ${VERSION}\\n**Release Notes:** ${RELEASE_URL}\\n\\n"
+          CONTENT+="📦 **Download:**\\n• [GitHub Release](${RELEASE_URL})\\n"
+          CONTENT+="• [PyPI Package](https://pypi.org/project/mcp-mesh/)\\n"
+          CONTENT+="• [Docker Images](https://hub.docker.com/u/mcpmesh)\\n"
+          CONTENT+="• [Homebrew](https://github.com/dhyansraj/homebrew-mcp-mesh)\\n\\n🎉 **What's New:**\\n"
+          CONTENT+="• macOS Native Builds - Intel & Apple Silicon support\\n• Homebrew Distribution - Official tap available\\n"
+          CONTENT+="• Enhanced PATH Resolution - Works with system installations\\n• Cross-Platform Binary Consistency"
+
           PAYLOAD=$(cat <<EOF
           {
-            "content": "🚀 **New MCP Mesh Release!**\n\n**Version:** ${VERSION}\n**Release Notes:** ${RELEASE_URL}\n\n📦 **Download:**\n• [GitHub Release](${RELEASE_URL})\n• [PyPI Package](https://pypi.org/project/mcp-mesh/)\n• [Docker Images](https://hub.docker.com/u/mcpmesh)\n\n🎉 **What's New:**\n• FastMCP Client Integration - Official protocol support with better compliance\n• Unified Telemetry Architecture - Complete observability across MCP agents and API services\n• FastAPI Route Telemetry - Distributed tracing for @mesh.route decorated endpoints\n• Agent ID Consistency - Unified generation eliminates registry/telemetry mismatch",
+            "content": "${CONTENT}",
             "thread_name": "MCP Mesh Release ${VERSION}"
           }
           EOF

--- a/.github/workflows/test-macos-build.yml
+++ b/.github/workflows/test-macos-build.yml
@@ -1,0 +1,138 @@
+name: Test macOS Build
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version (e.g., v0.5.2-test)"
+        required: true
+        default: "v0.5.2-test"
+
+env:
+  GO_VERSION: "1.23"
+
+jobs:
+  # Build Linux binaries for comparison
+  build-linux-binaries:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Install cross-compilation toolchain
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-aarch64-linux-gnu gcc-x86-64-linux-gnu
+
+      - name: Build Linux binaries
+        env:
+          VERSION: ${{ github.event.inputs.version }}
+          PLATFORMS: "linux/amd64,linux/arm64"
+          OUTPUT_DIR: "dist-linux"
+        run: ./packaging/scripts/build-binaries.sh
+
+      - name: Upload Linux artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: linux-binaries-${{ github.event.inputs.version }}
+          path: |
+            dist-linux/*.tar.gz
+            dist-linux/*.zip
+
+  # Build macOS binaries natively
+  build-macos-binaries:
+    runs-on: macos-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Build macOS binaries
+        env:
+          VERSION: ${{ github.event.inputs.version }}
+          PLATFORMS: "darwin/amd64,darwin/arm64"
+          OUTPUT_DIR: "dist-macos"
+        run: ./packaging/scripts/build-binaries.sh
+
+      - name: Upload macOS artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: macos-binaries-${{ github.event.inputs.version }}
+          path: |
+            dist-macos/*.tar.gz
+            dist-macos/*.zip
+
+  # Verify both builds completed successfully
+  verify-builds:
+    runs-on: ubuntu-latest
+    needs: [build-linux-binaries, build-macos-binaries]
+    permissions:
+      contents: read
+    steps:
+      - name: Download Linux binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: linux-binaries-${{ github.event.inputs.version }}
+          path: linux-artifacts
+
+      - name: Download macOS binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: macos-binaries-${{ github.event.inputs.version }}
+          path: macos-artifacts
+
+      - name: List all generated artifacts
+        run: |
+          echo "=== Linux Artifacts ==="
+          find linux-artifacts -type f -name "*.tar.gz" -o -name "*.zip" | sort
+
+          echo "=== macOS Artifacts ==="
+          find macos-artifacts -type f -name "*.tar.gz" -o -name "*.zip" | sort
+
+          echo "=== File Sizes ==="
+          echo "Linux binaries:"
+          find linux-artifacts -type f -exec ls -lh {} \;
+          echo "macOS binaries:"
+          find macos-artifacts -type f -exec ls -lh {} \;
+
+      - name: Generate test report
+        run: |
+          echo "# macOS Build Test Results" > test-report.md
+          echo "" >> test-report.md
+          echo "## Generated Artifacts" >> test-report.md
+          echo "" >> test-report.md
+          echo "### Linux (Cross-compiled)" >> test-report.md
+          find linux-artifacts -name "*.tar.gz" -exec echo "- {}" \; >> test-report.md
+          echo "" >> test-report.md
+          echo "### macOS (Native)" >> test-report.md
+          find macos-artifacts -name "*.tar.gz" -exec echo "- {}" \; >> test-report.md
+          echo "" >> test-report.md
+          echo "## Next Steps" >> test-report.md
+          echo "1. Download macOS artifacts" >> test-report.md
+          echo "2. Test install script on macOS" >> test-report.md
+          echo "3. Verify meshctl and registry work correctly" >> test-report.md
+
+          cat test-report.md
+
+      - name: Upload test report
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-report-${{ github.event.inputs.version }}
+          path: test-report.md

--- a/install.sh
+++ b/install.sh
@@ -250,23 +250,25 @@ install_binary() {
     # Use the same platform format as the release assets (with underscores)
     local platform_for_path="${PLATFORM//-/_}"
     if [[ "$binary_name" == "registry" ]]; then
-        extracted_binary="$temp_dir/$platform_for_path/registry"
+        extracted_binary="$temp_dir/$platform_for_path/mcp-mesh-registry"
+        install_name="mcp-mesh-registry"
     else
         extracted_binary="$temp_dir/$platform_for_path/$binary_name"
+        install_name="$binary_name"
     fi
 
     if [[ -w "$INSTALL_DIR" ]]; then
-        cp "$extracted_binary" "$INSTALL_DIR/$binary_name"
-        chmod +x "$INSTALL_DIR/$binary_name"
+        cp "$extracted_binary" "$INSTALL_DIR/$install_name"
+        chmod +x "$INSTALL_DIR/$install_name"
     else
-        sudo cp "$extracted_binary" "$INSTALL_DIR/$binary_name"
-        sudo chmod +x "$INSTALL_DIR/$binary_name"
+        sudo cp "$extracted_binary" "$INSTALL_DIR/$install_name"
+        sudo chmod +x "$INSTALL_DIR/$install_name"
     fi
 
     # Cleanup
     rm -rf "$temp_dir"
 
-    print_info "✅ $binary_name installed successfully!"
+    print_info "✅ $install_name installed successfully!"
 }
 
 # Install meshctl
@@ -287,11 +289,11 @@ install_registry() {
     install_binary "registry"
 
     # Verify installation
-    if command -v registry >/dev/null 2>&1; then
-        print_info "Verification: registry installed to $INSTALL_DIR/registry"
+    if command -v mcp-mesh-registry >/dev/null 2>&1; then
+        print_info "Verification: mcp-mesh-registry installed to $INSTALL_DIR/mcp-mesh-registry"
     else
-        print_warn "registry was installed to $INSTALL_DIR but is not in PATH"
-        print_warn "Add $INSTALL_DIR to your PATH or use the full path: $INSTALL_DIR/registry"
+        print_warn "mcp-mesh-registry was installed to $INSTALL_DIR but is not in PATH"
+        print_warn "Add $INSTALL_DIR to your PATH or use the full path: $INSTALL_DIR/mcp-mesh-registry"
     fi
 }
 
@@ -333,7 +335,7 @@ main() {
             print_info "Run 'meshctl --help' to get started with the CLI"
         fi
         if [[ "$INSTALL_REGISTRY" == "true" ]]; then
-            print_info "Run 'registry --help' to get started with the registry"
+            print_info "Run 'mcp-mesh-registry --help' to get started with the registry"
         fi
     fi
 }

--- a/packaging/scripts/build-binaries.sh
+++ b/packaging/scripts/build-binaries.sh
@@ -245,7 +245,7 @@ build_all_binaries() {
         # Build meshctl
         if build_binary "meshctl" "$goos" "$goarch" "meshctl"; then
             # Build registry
-            if build_binary "registry" "$goos" "$goarch" "registry"; then
+            if build_binary "registry" "$goos" "$goarch" "mcp-mesh-registry"; then
                 built_platforms+=("${goos}_${goarch}")
             else
                 failed_builds+=("registry-${goos}_${goarch}")

--- a/src/core/cli/process_lifecycle.go
+++ b/src/core/cli/process_lifecycle.go
@@ -253,29 +253,10 @@ func (pm *ProcessManager) StartRegistryProcess(port int, dbPath string, metadata
 		dbPath = pm.config.DBPath
 	}
 
-	// Start registry process - try multiple locations for registry binary
-	possiblePaths := []string{
-		"./bin/mcp-mesh-registry",           // Preferred location (matches Makefile)
-		"./mcp-mesh-registry",               // Legacy location
-		"./build/mcp-mesh-registry",         // Build directory
-		"cmd/mcp-mesh-registry/mcp-mesh-registry", // Legacy build location
-		"mcp-mesh-registry",                 // In PATH
-	}
-
-	var registryBinary string
-	var binaryFound bool
-
-	// Check each possible location
-	for _, path := range possiblePaths {
-		if _, err := os.Stat(path); err == nil {
-			registryBinary = path
-			binaryFound = true
-			break
-		}
-	}
-
-	if !binaryFound {
-		return nil, fmt.Errorf("registry binary not found at any of these locations: %v. Please ensure the binary is built or run 'make build' to compile it", possiblePaths)
+	// Find registry binary
+	registryBinary, err := pm.findRegistryBinary()
+	if err != nil {
+		return nil, fmt.Errorf("failed to start registry: %w", err)
 	}
 
 	args := []string{
@@ -393,8 +374,12 @@ func (pm *ProcessManager) startProcessInternal(name string, info *ProcessInfo) e
 	var cmd *exec.Cmd
 
 	if info.ServiceType == "registry" {
-		// Start registry process
-		registryBinary := "./mcp-mesh-registry"
+		// Find registry binary
+		registryBinary, err := pm.findRegistryBinary()
+		if err != nil {
+			return err
+		}
+
 		port := 8080
 		if portVal, ok := info.Metadata["port"].(int); ok {
 			port = portVal
@@ -433,6 +418,32 @@ func (pm *ProcessManager) startProcessInternal(name string, info *ProcessInfo) e
 	info.LastSeen = time.Now()
 
 	return nil
+}
+
+// findRegistryBinary finds the registry binary in local paths or PATH
+func (pm *ProcessManager) findRegistryBinary() (string, error) {
+	// First check local paths
+	localPaths := []string{
+		"./bin/mcp-mesh-registry",
+		"./mcp-mesh-registry",
+		"./build/mcp-mesh-registry",
+		"cmd/mcp-mesh-registry/mcp-mesh-registry",
+	}
+
+	for _, path := range localPaths {
+		if _, err := os.Stat(path); err == nil {
+			return path, nil
+		}
+	}
+
+	// If not found locally, check PATH
+	if path, err := exec.LookPath("mcp-mesh-registry"); err == nil {
+		return path, nil
+	}
+
+	// Return error with all attempted paths
+	allPaths := append(localPaths, "mcp-mesh-registry (in PATH)")
+	return "", fmt.Errorf("registry binary not found at any of these locations: %v. Please ensure the binary is built or run 'make build' to compile it", allPaths)
 }
 
 // generateProcessName generates a unique process name from a file path

